joint_limits/include/joint_limits/joint_limiter_interface.hpp

```diff
@@ -130,6 +132,31 @@ class JointLimiterInterface
     return result;
   }
 
+  /**
+   * Wrapper init method that accepts the joint names and their limits directly
+   */
+  JOINT_LIMITS_PUBLIC virtual bool init(
+    const std::vector<std::string> & joint_names, const std::vector<LimitsType> & joint_limits,
+    const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr & param_itf,
+    const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr & logging_itf)
+  {
+    number_of_joints_ = joint_names.size();
+    joint_names_ = joint_names;
+    joint_limits_ = joint_limits;
+    node_param_itf_ = param_itf;
+    node_logging_itf_ = logging_itf;
+    updated_limits_.writeFromNonRT(joint_limits_);
+
+    if ((number_of_joints_ != joint_limits_.size()) && has_logging_interface())
```

Comment:
I can change it to use it properly. This is a new `init` method and this will take a vector of LimitsType and you will need to check if this size is good or not w.r.t joint_names size